src/jdk.dynalink/share/classes/jdk/dynalink/DynamicLinkerFactory.java

```diff
@@ -470,11 +458,8 @@ private List<GuardingDynamicLinker> discoverAutoLoadLinkers() {
         return discovered;
     }
 
-    @SuppressWarnings("removal")
     private static ClassLoader getThreadContextClassLoader() {
-        return AccessController.doPrivileged(
-            (PrivilegedAction<ClassLoader>) () -> Thread.currentThread().getContextClassLoader(),
-            GET_CLASS_LOADER_CONTEXT);
+        return Thread.currentThread().getContextClassLoader();
```

Comment:
This could get inlined. It's only invoked from a single location, and the sole purpose of it was to evaluate the expression in doPrivileged.